src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java

```diff
@@ -55,6 +56,12 @@ public void onPlayerChangeWorld(PlayerChangedWorldEvent e){
         createOrUpdateTeam(e.getPlayer());
     }
 
+    @EventHandler
+    public void onPlayerQuit(PlayerQuitEvent e) {
+        PacketManager.getInstance().removeListener(collisionPacketListener, e.getPlayer());
+        playerTeamMap.remove(e.getPlayer());
```

Comment:
Not strictly needed, as that is a [WeakHashMap](https://github.com/kernitus/BukkitOldCombatMechanics/blob/59ecdc25d9d6e59a20d9872f4f1a7a82fade40c9/src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerCollisions.java#L32).

Reply:
It is always best practice to clean collections properly, you are still keeping references to TeamPacket objects.

Reply:
> Hash table based implementation of the Map interface, with weak keys. An entry in a WeakHashMap will automatically be removed when its key is no longer in ordinary use. More precisely, the presence of a mapping for a given key will not prevent the key from being discarded by the garbage collector, that is, made finalizable, finalized, and then reclaimed. **When a key has been discarded its entry is effectively removed from the map**, so this class behaves somewhat differently from other Map implementations. 

Where does the reference to the TeamPacket object survive?

Reply:
hmm, fair

Reply:
It is probably still nice to have it though. As we just saw that code might be a bit too clever for its own good.